### DISCUSSION TITLE
Support for kerberos authentication

### DISCUSSION THREAD

**Discussion by Simhan-dev** [kubernetes-csi/csi-driver-nfs#380] [Q&A]
Hi, 
I have a requirement where i need to mount an external data source via nfs v4 to the k8s cluster. I need to be able to provide krb5 authentication. Is there a way to specify krb5 authentication in the csi driver?

**Comment by andyzhangx** [comment]
what's the regular mount options for krb5 authentication? like `mount -t nfs -o ...`

**Reply by Flightkick** [reply]
The mount options contain a `sec` parameter, which can be one of `krb5`, `krb5i`, `krb5p`.

An example would be `-o "rw,sec=krb5p,vers=4.1"`

I'm not too familiar with CSI architecture but I expect the node job to require access to the Kerberos keytab and the driver image to provide client libraries.

**Reply by andyzhangx** [reply]
@Flightkick is this sth. we could implement same as smb csi driver kerberos support? https://github.com/kubernetes-csi/csi-driver-smb/blob/master/docs/driver-parameters.md#kerberos-ticket-support-for-linux

**Reply by Flightkick** [reply]
Oh, that might be useful!

This plugin is targeted against Kubernetes but I'm using it in combination with Nomad.  I'm not sure to which degree this would be compatible with Nomad, I don't expect support for that but in case it does work, it would be nice.
